### PR TITLE
Check for invalid input

### PR DESCRIPTION
**Pull request by Mottie**
Throw error on invalid obj parameter.

### CODE REVIEW

**Comment by Mottie** [comment]
The failing checks are part of the optimization tests that failed for me even before I made these changes.

File: index.js [hunk 1]
```diff
@@ -57,13 +57,18 @@ module.exports = (obj, opts) => {
 		promiseModule: Promise
 	}, opts);
 
+	const objType = typeof obj;
+	if (objType === 'string' || objType === 'undefined' || obj === null) {
+		throw new TypeError(`Invalid object type "${obj === null ? 'null' : objType}"`);
```
Comment (kevva) [inline index.js]:
```js
throw new TypeError(`Expected \`input`\ to be a \`Function\` or \`Object\`, got \`${obj === null ? 'null' : objType}\``);
```

File: .gitignore [hunk 2]
```diff
@@ -1 +1,3 @@
 node_modules
+yarn.lock
+package-lock.json
```
Comment (kevva) [inline .gitignore]: These are unrelated to this PR.
